outbox-core/src/main/java/io/praesidio/outbox/MessageRelay.java

```diff
@@ -0,0 +1,41 @@
+package io.praesidio.outbox;
+
+import io.praesidio.outbox.values.MessageType;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class MessageRelay {
+
+    private final MessageRepository messageRepository;
+    private final Map<MessageType, MessageRelayProvider> messageRelayProviders;
+    private final Map<MessageType, MessageSerializer> messageSerializers;
+
+    public MessageRelay(
+            MessageRepository messageRepository,
+            Set<MessageRelayProvider> messageRelayProviders,
+            Set<MessageSerializer> messageSerializers
+    ) {
+        this.messageRepository = messageRepository;
+        this.messageRelayProviders = messageRelayProviders.stream().collect(Collectors.toMap(MessageRelayProvider::getType, s -> s));
```

Comment:
Should we care about duplicates?

I mean two serializers/providers for the same type.

Reply:
I created an issue: #12